src/components/compareImages/CompareImages.tsx

```diff
@@ -0,0 +1,103 @@
+import React, { FC, useEffect, useRef, useState } from 'react';
+
+import styles from './CompareImages.css';
+
+interface CompareImagesProps {
+  leftPhoto: string;
+  rightPhoto: string;
+}
+
+const CompareImages: FC<CompareImagesProps> = ({ leftPhoto, rightPhoto }) => {
+  const [isDown, setIsDown] = useState<boolean>(false);
+
+  const lineRef = useRef<HTMLDivElement>(null);
+  const wrapperRef = useRef<HTMLDivElement>(null);
+  const rightPhotoRef = useRef<HTMLDivElement>(null);
+
+  const downHandler = () => setIsDown(true);
+  const upHandler = () => setIsDown(false);
+
+  useEffect(() => {
+    const { current: lineElement } = lineRef;
+    const { current: wrapperElement } = wrapperRef;
+    const { current: rightPhoto } = rightPhotoRef;
+
+    if (!(wrapperElement && lineElement && rightPhoto)) {
+      return;
+    }
+
+    const leftBorder = wrapperElement.getBoundingClientRect().left;
+    const wrapperWidth = wrapperElement.getBoundingClientRect().width;
+
+    const getNewX = (x: number) => {
+      let correctX = x - leftBorder;
+
+      correctX = correctX <= 0 ? 0 : correctX;
+      correctX = correctX >= wrapperWidth ? wrapperWidth : correctX;
+
+      return correctX;
+    };
+
+    const getPercent = (x: number) => {
+      return (x / wrapperWidth) * 100;
+    };
```

Comment:
please move `getNewX` and `getPercent` to component utils file, they can be easily transformed to small pure functions